src/ssl.c

```diff
@@ -12708,8 +12708,19 @@ static int wolfSSL_parse_cipher_list(WOLFSSL_CTX* ctx, Suites* suites,
 
 #ifdef WOLFSSL_SMALL_STACK
     suitesCpy = (byte*)XMALLOC(suites->suiteSz, NULL, DYNAMIC_TYPE_TMP_BUFFER);
-    if (suitesCpy == NULL)
-        return WOLFSSL_FAILURE;
+    if (suitesCpy == NULL) {
```

Comment:
This seems a little odd to me, that if `suites->suiteSz` is 0 that we would allocate `suitesCpy` to a size of `1`.

Is this the correct way to handle this, or should we just be skipping the `XMALLOC` all together if `suites->suiteSz` is 0?  Then adjusting logic inside this function to accomodate.

Reply:
That's true. I think that just skipping the XMALLOC when `suites->suiteSz` is zero is better than the logic here. will check that with CC-RX